src/Features/Core/Portable/DesignerAttributes/AbstractDesignerAttributeService.cs

```diff
@@ -32,30 +32,33 @@ public async Task<DesignerAttributeResult> ScanDesignerAttributesAsync(Document
             // make sure given input is right one
             Contract.ThrowIfFalse(_workspace == document.Project.Solution.Workspace);
 
-            // same service run in both inproc and remote host, but remote host will not have RemoteHostClient service, 
-            // so inproc one will always run
-            var client = await document.Project.Solution.Workspace.TryGetRemoteHostClientAsync(cancellationToken).ConfigureAwait(false);
-            if (client != null && !document.IsOpen())
+            // run designer attributes scanner on remote host
+            // we only run closed files to make open document to have better responsiveness. 
+            // also we cache everything related to open files anyway, no saving by running
+            // them in remote host
+            if (!document.IsOpen())
             {
-                // run designer attributes scanner on remote host
-                // we only run closed files to make open document to have better responsiveness. 
-                // also we cache everything related to open files anyway, no saving by running
-                // them in remote host
-                return await ScanDesignerAttributesInRemoteHostAsync(client, document, cancellationToken).ConfigureAwait(false);
+                var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
+                if (client != null)
+                {
+                    var result = await client.TryRunRemoteAsync<DesignerAttributeResult>(
+                        WellKnownServiceHubServices.CodeAnalysisService,
+                        nameof(IRemoteDesignerAttributeService.ScanDesignerAttributesAsync),
+                        document.Project.Solution,
```

Comment:
TryGetClientAsync just checks if the project language supports remote services, the client does not hold on a solution:

```C#
if (!RemoteSupportedLanguages.IsSupported(project.Language))
```

Presumably this is to avoid calling to remote host for TypeScript and FSharp projects.